components/script/dom/htmltablerowelement.rs

```diff
@@ -54,6 +56,14 @@ impl HTMLTableRowElement {
                            document,
                            HTMLTableRowElementBinding::Wrap)
     }
+
+    /// Determine the index for this `HTMLTableRowElement` within the given
+    /// `HTMLCollection`. Returns `-1` if not found within collection.
+    fn row_index(&self, collection: Root<HTMLCollection>) -> i32 {
```

Comment:
Since this is a native Rust fn, the more idiomatic way would be returning an Option<i32> instead of returning -1 for the not found case.